dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -748,6 +752,54 @@ public static VulkanRMWOp newRMWOp(Expression address, Register register, Expres
                                            IntBinaryOp op, String mo, String scope) {
             return new VulkanRMWOp(register, address, op, value, mo, scope);
         }
+
+        public static VulkanRMWExtremum newRMWExtremum(Expression address, Register register, IntCmpOp op,
+                                                       Expression value, String mo, String scope) {
+            return new VulkanRMWExtremum(register, address, op, value, mo, scope);
+        }
+
+        public static VulkanCmpXchg newVulkanCmpXchg(Expression address, Register register, Expression expected,
+                                                     Expression value, String mo, String scope) {
+            return new VulkanCmpXchg(register, address, expected, value, mo, scope);
+        }
+    }
+
+    // =============================================================================================
+    // =========================================== Spir-V ==========================================
+    // =============================================================================================
+
+    public static class Spirv {
+        private Spirv() {}
+
+        public static SpirvLoad newSpirvLoad(Register register, Expression address, String scope,
+                                             Set<String> tags) {
+            return new SpirvLoad(register, address, scope, tags);
+        }
+
+        public static SpirvStore newSpirvStore(Expression address, Expression value, String scope,
+                                               Set<String> tags) {
+            return new SpirvStore(address, value, scope, tags);
+        }
+
+        public static SpirvXchg newSpirvXchg(Register register, Expression address, Expression value,
+                                             String scope, Set<String> tags) {
+            return new SpirvXchg(register, address, value, scope, tags);
+        }
+
+        public static SpirvRmw newSpirvRmw(Register register, Expression address, IntBinaryOp op, Expression value,
+                                            String scope, Set<String> tags) {
+            return new SpirvRmw(register, address, op, value, scope, tags);
+        }
+
+        public static SpirvCmpXchg newSpirvCmpXchg(Register register, Expression address, Expression cmp, Expression value,
+                                                   String scope, Set<String> eqTags, Set<String> neqTags) {
+            return new SpirvCmpXchg(register, address, cmp, value, scope, eqTags, neqTags);
+        }
+
+        public static SpirvRmwExtremum newSpirvRmwExtremum(Register register, Expression address, IntCmpOp op, Expression value,
+                                                           String scope, Set<String> tags) {
+            return new SpirvRmwExtremum(register, address, op, value, scope, tags);
+        }
```

Comment:
Keep in mind that the events might simply exists for the fact that the Compilation pass can visit them properly.
I think the `EventVisitor` interface starts to get too bloated with new language support. I believe this can be fixed very nicely though, by having visitors for each frontend language (e.g. `SpirvEventVisitor`) and have those language-specific events implement their accept method by 
```Java
if (visitor instanceof SpirvEventVisitor spirvVis) 
    return spirvVis.visitSpirvXYZEvent(this);
else 
    return visitor.visitGeneralEvent(this);
```
Now the Spirv package could define the `SpirvCompilationVisitor` which implements `SpirvEventVisitor`. The parser could create an instance of that visitor and attach it to the program instance so that the Compilation pass can just use the instance rather than creating it on demand.
The advantage of this is that none of the core code needs to know about Spirv events, not even the `EventFactory`. All Spirv events, tags, compilers etc. could be fully contained in the Spirv package.
The same can be done for all frontends.

I wouldn't suggest to add this solution into this PR, but I believe that we should eventually do it cleanly remove frontend code from the core.

Reply:
In my opinion this EventFactory is useless in general, but I added it here to be consistent with the rest of the codebase.

Reply:
The question was not about the factory but rather the existence of all new events. But there is another part where you commented on this. Lets keep the discussion in that one.